### PR TITLE
Docs, part 2

### PR DESCRIPTION
**Pull request by arogachev**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ❌
| Breaks BC?    | ❌
| Fixed issues  | #358

- Add docs for `SkipOnEmptyInterface` and `SkipOnEmptyTrait`.
- Fix ambiguity in `SkipOnEmptyTrait::getSkipOnEmptyOption()` when handling callable.

### CODE REVIEW

File: src/SkipOnEmptyInterface.php [hunk 1]
```diff
@@ -4,9 +4,96 @@
...
+     *     return $this->skipOnEmpty;
+     * }
+     * ```
+     *
+     * During pre-validation phase it will be normalized to an "empty criteria" - a callable identifying when and which
+     * values exactly must be considered as empty and skipped or not skipped at all.
+     *
+     * @return bool|callable|null A raw value set in the constructor:
+     *
+     * - `null` and `false` - never skip a rule (the validated value is always considered as not empty). Matching
```
Comment (vjik) [inline src/SkipOnEmptyInterface.php:65]: Null - will be used default empty criteria from validator.

**Review by vjik — CHANGES_REQUESTED** [review]

File: src/SkipOnEmptyInterface.php [hunk 2]
```diff
@@ -4,9 +4,96 @@
...
 namespace Yiisoft\Validator;
 
+use Yiisoft\Validator\EmptyCriteria\NeverEmpty;
+use Yiisoft\Validator\EmptyCriteria\WhenEmpty;
+use Yiisoft\Validator\EmptyCriteria\WhenNull;
+use Yiisoft\Validator\Rule\Trait\SkipOnEmptyTrait;
+
+/**
+ * An optional interface for rules to implement. It allows skipping validation for a rule when the validated value is
+ * "empty". Requires an additional private class property `$skipOnEmpty`. In package rules it's `null` by default:
```
Comment (samdark) [inline src/SkipOnEmptyInterface.php:14]: > Requires an additional private class property `$skipOnEmpty`.

Interface can't require a property, especially private one.

Comment (samdark) [reply src/SkipOnEmptyInterface.php:14]: I think that belongs to `SkipOnEmptyTrait` description and not here.

File: src/SkipOnEmptyInterface.php [hunk 3]
```diff
@@ -4,9 +4,96 @@
...
+ * }
+ * ```
+ *
+ * The package ships with {@see SkipOnEmptyTrait} which already implements that interface. All you have to do is include
+ * it in the rule class.
+ */
 interface SkipOnEmptyInterface
 {
+    /**
+     * A setter to change `$skipOnEmpty` property. Must be implemented following immutability concept:
```
Comment (samdark) [inline src/SkipOnEmptyInterface.php:31]: Interface knows nothing about how the value is stored so we should not mention implementation details.

Comment (samdark) [reply src/SkipOnEmptyInterface.php:31]: Immuntability could be mentioned as a requirement, but I don't think an example is needed.

File: src/SkipOnEmptyInterface.php [hunk 4]
```diff
@@ -4,9 +4,96 @@
...
+     * ```
+     *
+     * @param bool|callable|null $value
+     *
+     * @return $this
+     */
     public function skipOnEmpty(bool|callable|null $value): static;
 
+    /**
+     * A getter for `$skipOnEmpty` property. Returned value is unchanged. Typical implementation is just a simple
```
Comment (samdark) [inline src/SkipOnEmptyInterface.php:50]: Same here. Should not mention properties. That's not part of the interface.

**Review by samdark — CHANGES_REQUESTED** [review]

**Comment by arogachev** [comment]
The part with default value in `SkipOnEmptyTrait::getSkipOnEmptyOption()` can be handled later - #411.

**Review by vjik — APPROVED** [review]

**Review by samdark — APPROVED** [review]

**Pull request merged**